packages/plugins/src/DocumentAssetsPlugin.ts

```diff
@@ -0,0 +1,174 @@
+import type { Page, Plugin as PluginType } from '@jpmorganchase/mosaic-types';
+import fsExtra from 'fs-extra';
+import glob from 'fast-glob';
+import path from 'path';
+import { escapeRegExp } from 'lodash-es';
+import { unified } from 'unified';
+import { visit } from 'unist-util-visit';
+import remarkParse from 'remark-parse';
+import remarkStringify from 'remark-stringify';
+import { VFile } from 'vfile';
+
+interface DocumentAssetsPluginOptions {
+  /**
+   * A collection of subdirectory names that contain assets, referenced by pages in the associated parent directory
+   * @default: ['images']
+   */
+  assetSubDirs?: string[];
+  /**
+   * The source path, where your docs reside, when the site runs
+   */
+  srcDir?: string;
+  /**
+   * The directory to copy matched assets to, typically the site's public directory
+   * @default './public'
+   */
+  outputDir?: string;
+  /**
+   * The prefix we add to all images in documents, so that it routes to the public directory
+   * @default '/images'
+   */
+  imagesPrefix?: string;
+}
+
+function isUrl(assetPath: string): boolean {
+  try {
+    new URL(assetPath);
+    return true;
+  } catch (_err) {}
+  return false;
+}
+
+function isRelativePath(assetPath: string): boolean {
+  const relativePathRegExp = /^(\.{1,2}\/)+/;
+  return relativePathRegExp.test(assetPath);
+}
+
+const createPageTest = (ignorePages: string[], pageExtensions: string[]) => {
+  const extTest = new RegExp(`${pageExtensions.map(ext => escapeRegExp(ext)).join('|')}$`);
+  const ignoreTest = new RegExp(`${ignorePages.map(ignore => escapeRegExp(ignore)).join('|')}$`);
+
+  return (file: string) =>
+    !ignoreTest.test(file) && extTest.test(file) && !path.basename(file).startsWith('.');
+};
+
+function remarkRewriteImagePaths(newPrefix: string) {
+  return (tree: any) => {
+    visit(tree, 'image', (node: any) => {
+      if (node.url) {
+        if (isUrl(node.url) || /^\//.test(node.url)) {
+          // Absolute URL or path, do nothing
+          return;
+        } else {
+          const assetPath = isRelativePath(node.url) ? node.url : `./${node.url}`;
+          const resolvedPath = path.resolve(path.dirname(newPrefix), assetPath);
+          node.url = resolvedPath;
+        }
+      }
+    });
+  };
+}
+
+/**
+ * Plugin that finds assets within the Mosaic filesystem and copies them to the configured `outputDir`.
+ * Documents that create relative references to those images, will be re-written to pull the images from the `outputDir`.
+ */
+const DocumentAssetsPlugin: PluginType<Page, DocumentAssetsPluginOptions> = {
+  async afterUpdate(
+    _mutableFileSystem,
+    _helpers,
+    {
+      assetSubDirs = ['images'],
```

Comment:
It could be a lot simpler to make this a string and to force globs to be provided e.g. assetDirs: "**/{images,assets}"

Reply:
Good shout.. Might make sense if combining sites through scopes and you don't want to copy all the images